packages/cactus-plugin-ledger-connector-besu/src/test/typescript/benchmark/run-plugin-ledger-connector-besu-benchmark.ts

```diff
@@ -0,0 +1,239 @@
+      __dirname,
+      "../../../../../../", // walk back up to the project root
+    );
+
+    log.info("%s gitRootPath=%s", LOG_TAG, gitRootPath);
+
+    const DEFAULT_OUTPUT_FILE_RELATIVE_PATH =
+      ".tmp/benchmark-results/plugin-ledger-connector-besu/run-plugin-ledger-connector-besu-benchmark.ts.log";
+
+    const relativeOutputFilePath =
+      opts.argv[2] === undefined
+        ? DEFAULT_OUTPUT_FILE_RELATIVE_PATH
+        : opts.argv[2];
+
+    log.info(
+      "%s DEFAULT_OUTPUT_FILE_RELATIVE_PATH=%s",
+      LOG_TAG,
+      DEFAULT_OUTPUT_FILE_RELATIVE_PATH,
+    );
+
+    log.info("%s opts.argv[2]=%s", LOG_TAG, opts.argv[2]);
+
+    log.info("%s relativeOutputFilePath=%s", LOG_TAG, relativeOutputFilePath);
+
+    const absoluteOutputFilePath = path.join(
+      gitRootPath,
+      relativeOutputFilePath,
+    );
+
+    log.info("%s absoluteOutputFilePath=%s", LOG_TAG, absoluteOutputFilePath);
+
+    const absoluteOutputDirPath = path.dirname(absoluteOutputFilePath);
+    log.info("%s absoluteOutputDirPath=%s", LOG_TAG, absoluteOutputDirPath);
+
+    await fse.mkdirp(absoluteOutputDirPath);
+    log.info("%s mkdir -p OK: %s", LOG_TAG, absoluteOutputDirPath);
+
+    const minSamples = 100;
+    const suite = new Benchmark.Suite({});
+
+    const cycles: string[] = [];
+
+    await new Promise((resolve, reject) => {
+      suite
+        .add("plugin-ledger-connector-besu_HTTP_GET_getOpenApiSpecV1", {
+          defer: true,
+          minSamples,
+          fn: async function (deferred: Benchmark.Deferred) {
+            await httpApi.getOpenApiSpecV1();
+            deferred.resolve();
+          },
+        })
+        .on("cycle", (event: { target: unknown }) => {
+          // Output benchmark result by converting benchmark result to string
+          // Example line on stdout:
+          // plugin-ledger-connector-besu_HTTP_GET_getOpenApiSpecV1 x 1,020 ops/sec ±2.25% (177 runs sampled)
+          const cycle = String(event.target);
+          log.info("%s Benchmark.js CYCLE: %s", LOG_TAG, cycle);
+          cycles.push(cycle);
+        })
+        .on("complete", function () {
+          log.info("%s Benchmark.js COMPLETE.", LOG_TAG);
+          resolve(suite);
+        })
+        .on("error", async (ex: unknown) => {
+          log.info("%s Benchmark.js ERROR: %o", LOG_TAG, ex);
+          reject(ex);
+        })
+        .run();
+    });
+
+    const data = cycles.join(EOL);
+    log.info("%s Writing results...", LOG_TAG);
+    await fse.writeFile(absoluteOutputFilePath, data, { encoding: "utf-8" });
+    log.info("%s Wrote results to %s", LOG_TAG, absoluteOutputFilePath);
+  } finally {
+    await apiServer.shutdown();
+    log.info("%s Shut down API server OK", LOG_TAG);
+
+    await besuTestLedger.stop();
+    await besuTestLedger.destroy();
+  }
+};
+
+main({ argv: process.argv })
+  .then(async () => {
+    console.log("%s Script execution completed successfully", LOG_TAG);
+    process.exit(1);
```

Comment:
@ruzell22 I tested the script and it appears to be working fine, so LGTM on that in general, BUT here you are exiting with code 1 which signals to the operating system that for some reason the execution has failed. Please exit with code zero instead which means success to the operating system.